kernel/src/process.rs

```diff
@@ -333,8 +337,12 @@ pub enum Task {
 ///
 /// Likely these four arguments will get passed as the first four register
 /// values, but this is architecture-dependent.
+///
+/// A `FunctionCall` also identifies the callback that scheduled it, if any, so
+/// that it can be unscheduled when the process unsubscribes from this callback.
 #[derive(Copy, Clone, Debug)]
 pub struct FunctionCall {
+    pub callback_id: Option<CallbackId>,
```

Comment:
I'm not sure I agree with myself, but i think using `Option` makes it less clear what circumstances might warrant the `None` case. What about a separate enum, something like:

```rust
enum CallbackSource {
    Kernel, // For, e.g. `init_fn`
    Driver(CallbackId)
}
```

Reply:
Yes that totally makes sense! I named it `FunctionCallSource` though, as `init_fn` is not a callback.